### DISCUSSION TITLE
cImport: how to add more includepath?

### DISCUSSION THREAD

**Discussion by kassane** [zigtools/zls#636] [Q&A] [answered]
Just like the one described in the title. Only the libraries installed on the system are recognized by zls.
Is it possible for zls to be based on the headers included in `build.zig`?

**Comment by techatrix** [comment, marked as answer]
zls automatically runs your `build.zig` and extracts all include dirs which are added with `addIncludePath`.

I would recommend doing the following things to possibly find the issue:
1. try updating to the latest master version
2. double check if your `build.zig` adds all your required include dirs with `addIncludePath`
3. check if the files are actually located in those directories
4. Are there any error messages in your zls logs?  Maybe they indicate what the problem is. Where you find the logs of your language server depends on your code editor.
5. just post your entire zls logs so i can take a look at them. (you could also just skip to this step)

## How to open ZLS logs for some editors
| Code Editor   | Instructions                                   | Notes                                           |
| ------------- | ---------------------------------------------- | ----------------------------------------------- |
| VSCode        | Bottom Panel -> Output -> Zig Language Server  | requires the `augusterame.zls-vscode` extension |
| Sublime Text  | Command `LSP: Toggle Log Panel`                | requires the `LSP` package                      |
| Helix         | Command `Open the helix log file.`             |                                                 |
